pages/api/patients/create.ts

```diff
@@ -0,0 +1,48 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from 'next'
+import prisma from '../../../lib/prisma'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === 'POST') {
+    try {
+      let { user_id, pickup_enabled, dob, prescriptions } = req.body.data
+
+      var patient = null
+      if (!prescriptions) {
+        patient = await prisma.patient.create({
+          data: {
+            pickupEnabled: pickup_enabled,
+            dob: dob,
+            user: {
+              connect: {
+                id: user_id
+              }
+            }
+          }
+        })
+      } else {
+        patient = await prisma.patient.create({
+          data: {
+            pickupEnabled: pickup_enabled,
+            dob: dob,
+            user: {
+              connect: {
+                id: user_id
+              }
+            },
+            prescriptions: prescriptions
+          }
+        })
+      }
```

Comment:
Do we need to have two separate logic blocks for this? can we just set it to empty if there are no prescriptions that exist? lmk if I am missing something here!

```suggestion
        patient = await prisma.patient.create({
          data: {
            pickupEnabled: pickup_enabled,
            dob: dob,
            user: {
              connect: {
                id: user_id
              }
            },
            prescriptions: prescriptions ?? [ ]
          }
        })
```

Reply:
I tried this before, it errors on an empty list which is why i added the conditional, but ya i agree v weird

Reply:
I think it's unnecessary to pass in prescription on patient creation. Since the patient doesn't really have a prescription field in the database, but rather prescriptions have a patientId field. So when creating/assigning prescriptions it makes more sense to have patient creation, and then prescription creation separated.

Reply:
> I think it's unnecessary to pass in prescription on patient creation. Since the patient doesn't really have a prescription field in the database, but rather prescriptions have a patientId field. So when creating/assigning prescriptions it makes more sense to have patient creation, and then prescription creation separated.

That is a good point. I think that would be a better way

Reply:
I agree, will make change